ethstorage/miner/l1_mining_api.go

```diff
@@ -152,21 +109,55 @@ func (m *l1MiningAPI) SubmitMinedResult(ctx context.Context, contract common.Add
 		return common.Hash{}, fmt.Errorf("failed to estimate gas: %w", err)
 	}
 	m.lg.Info("Estimated gas done", "gas", estimatedGas)
-	cost := new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), gasPrice)
 	reward, err := m.GetMiningReward(rst.startShardId, rst.blockNumber.Int64())
 	if err != nil {
 		m.lg.Error("Query mining reward failed", "error", err.Error())
 		return common.Hash{}, err
 	}
-	profit := new(big.Int).Sub(reward, cost)
-	m.lg.Info("Estimated reward and cost (in ether)", "reward", weiToEther(reward), "cost", weiToEther(cost), "profit", weiToEther(profit))
-	if profit.Cmp(cfg.MinimumProfit) == -1 {
-		m.lg.Warn("Will drop the tx: the profit will not meet expectation",
-			"profitEstimated", weiToEther(profit),
-			"minimumProfit", weiToEther(cfg.MinimumProfit),
-		)
+	isProfitable := func(gasPrice *big.Int) bool {
+		cost := new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), gasPrice)
+		profit := new(big.Int).Sub(reward, cost)
+		m.lg.Debug("Estimating profit", "gasPrice", gasPrice)
+		m.lg.Debug("Estimated reward and cost (in ether)", "reward", fmtEth(reward), "cost", fmtEth(cost))
+		m.lg.Debug("Estimated profit (in ether)", "profit", fmtEth(profit), "minimumProfit", fmtEth(cfg.MinimumProfit))
+		return profit.Cmp(cfg.MinimumProfit) == 1
+	}
+
+	var useConfig bool
+	gasFeeCap := cfg.GasPrice
+	if gasFeeCap == nil || gasFeeCap.Cmp(common.Big0) == 0 {
+		gasFeeCap = new(big.Int).Add(baseFee, tip)
+	} else {
+		useConfig = true
+		m.lg.Info("Using configured gas price as gasFeeCap", "gasFeeCap", gasFeeCap)
+	}
+	if !isProfitable(gasFeeCap) {
+		m.lg.Warn("Will drop the tx: the profit will not meet expectation", "minimumProfit", fmtEth(cfg.MinimumProfit))
 		return common.Hash{}, errDropped
 	}
+
+	if !useConfig {
```

Comment:
A more straightforward approach is to calculate the gasFeeCap if !isProfitable(baseFee * 2 + tip)

Reply:
That has been considered but the concern is that the miner could miss chances to make profits by submitting more txes, since it is a rare case to close the deal with gas price of baseFee * 2 + tip after all.